test/regressions/fixtures/CssVarsProvider/MaterialUIDefaultDark.js

```diff
@@ -6,6 +6,12 @@ import Paper from '@mui/material/Paper';
 import Toolbar from '@mui/material/Toolbar';
 
 export default function MaterialUIDefaultDark() {
+  const [, rerender] = React.useState(false);
+  React.useEffect(() => {
+    // Trigger rerender to ensure that the UI does not change after the first render.
+    // To catch bug like https://github.com/mui/material-ui/issues/36452
+    rerender(true);
+  }, []);
```

Comment:
@mnajdova There is a flaw in this visual regression. It produces the correct snapshot if the component renders just once which happens in the visual regression test.

Added the rerender(true) to trigger the second render. This should cover the bug the this PR solves.